File: LeetCode/Programming Skills - Java/1822. Sign of the Product of an Array.java
class Solution {
    public int arraySign(int[] nums) {
        boolean positive = true;

        for (int i = 0; i < nums.length; i++) {
            if (nums[i] == 0) {
                return 0;
            } else if (nums[i] < 0) {
                positive = !positive;
            }
        }

        if (positive) {
            return 1;
        } else {
            return -1;
        }
    }
}

File: LeetCode/C#/1822. Sign of the Product of an Array.cs
public class Solution {
    public int ArraySign(int[] nums) {
        var neg = false;
        
        foreach(var num in nums)
        {
            if(num == 0)
            {
                return 0;
            }
            else if(num < 0)
            {
                neg = !neg;
            }
        }
        
        if(!neg)
        {
            return 1;
        }
        else
        {
            return -1;
        }
    }
}